block-storage/src/main/scala/coop/rchain/blockstorage/dag/codecs.scala

```diff
@@ -19,11 +19,11 @@ object codecs {
       byteString => ByteVector(byteString.toByteArray)
     )
 
-  val codecDeployId = xmapToByteString(variableSizeBytes(uint8, bytes))
+  val codecByteString = xmapToByteString(bytes)
```

Comment:
Could you elaborate why specifying size is not necessary? Is this change compatible with current node version or clean database is required?

Reply:
It's not compatible.

Size prefix is only needed if _bytes_ are combined together, like in a list or with multiple fields.

E.g. if bytes are `aaaa` and `bbbb` (pseudo representation) then decoding each will be correct, but just combine them `aaaabbbb` there is no way to distinguish what is the first part what is the second so something like `4aaaaa4bbbb` is needed for decoder to correctly recognize each piece.
This means we don't need to defined size prefix on individual codec but only when they are combined.

Reply:
> This means we don't need to defined size prefix on individual codec but only when they are combined.

Hm.. Maxim meet similar problem yesterday. I'll address him your comment.